SoftLayer/shell/core.py

```diff
@@ -0,0 +1,158 @@
+"""
+    SoftLayer.CLI.shell
+    ~~~~~~~~~~~~~~~~~~~
+    An interactive shell which exposes the CLI
+
+    :license: MIT, see LICENSE for more details.
+"""
+from __future__ import print_function
+from __future__ import unicode_literals
+import copy
+import os
+import shlex
+import sys
+import traceback
+
+import click
+from prompt_toolkit import completion as p_completion
+from prompt_toolkit import history as p_history
+from prompt_toolkit import shortcuts as p_shortcuts
+
+from SoftLayer.CLI import core
+from SoftLayer.CLI import environment
+
+# pylint: disable=broad-except
+
+ALL_ROUTES = [
+    ('exit', 'SoftLayer.shell.cmd_exit:cli'),
+    ('shell-help', 'SoftLayer.shell.cmd_help:cli'),
+    ('env', 'SoftLayer.shell.cmd_env:cli'),
+]
+
+ALL_ALIASES = {
+    '?': 'shell-help',
+    'help': 'shell-help',
+    'quit': 'exit',
+}
+
+
+class ShellExit(Exception):
+    """Exception raised to quit the shell."""
+    pass
+
+
+@click.command()
+@environment.pass_env
+@click.pass_context
+def cli(ctx, env):
+    """Enters a shell for slcli."""
+
+    # Set up prompt_toolkit settings
+    app_path = click.get_app_dir('softlayer')
+    if not os.path.exists(app_path):
+        os.makedirs(os.path.dirname(app_path))
```

Comment:
This should be
```python
    os.makedirs(app_path)
```

On mac I get the following. 
**os.path.dirname: /Users/christopher/Library/Application Support**
**app_path: /Users/christopher/Library/Application Support/softlayer**
```bash
~/Code/softlayer-python  $ ./slcli shell
Traceback (most recent call last):
  File "./slcli", line 7, in <module>
    load_entry_point('SoftLayer', 'console_scripts', 'slcli')()
  File "/Users/christopher/Code/softlayer-python/SoftLayer/CLI/core.py", line 178, in main
    cli.main(**kwargs)
  File "/Library/Python/2.7/site-packages/click-3.3-py2.7.egg/click/core.py", line 590, in main
    rv = self.invoke(ctx)
  File "/Library/Python/2.7/site-packages/click-3.3-py2.7.egg/click/core.py", line 936, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Python/2.7/site-packages/click-3.3-py2.7.egg/click/core.py", line 782, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Python/2.7/site-packages/click-3.3-py2.7.egg/click/core.py", line 416, in invoke
    return callback(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/click-3.3-py2.7.egg/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/Library/Python/2.7/site-packages/click-3.3-py2.7.egg/click/core.py", line 416, in invoke
    return callback(*args, **kwargs)
  File "/Users/christopher/Code/softlayer-python/SoftLayer/shell/core.py", line 54, in cli
    os.makedirs(os.path.dirname(app_path))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 17] File exists: '/Users/christopher/Library/Application Support'
```